### PR TITLE
Declare pihole.toml as UTF-8 document

### PR DESCRIPTION
**Pull request by DL6ER**
# What does this implement/fix?

Declare `pihole.toml` as UTF-8 document. we add a compile-time switch to use ASCII with UTF-8 escaping instead in case this is a necessity for anyone. We are in the third millennium A.C. but we also know people are still using Windows XP on the web...

**Related issue or feature (if applicable):** N/A

**Pull request in [docs](https://github.com/pi-hole/docs) with documentation (if applicable):** N/A

---
**By submitting this pull request, I confirm the following:** 

1. I have read and understood the [contributors guide](https://docs.pi-hole.net/guides/github/contributing/), as well as this entire template. I understand which branch to base my commits and Pull Requests against. 
2. I have commented my proposed changes within the code.
3. I am willing to help maintain this change if there are issues with it later.
4. It is compatible with the [EUPL 1.2 license](https://opensource.org/licenses/EUPL-1.1)
5. I have squashed any insignificant commits. ([`git rebase`](http://gitready.com/advanced/2009/02/10/squashing-commits-with-rebase.html))

## Checklist:

- [x] The code change is tested and works locally.
- [x] I based my code and PRs against the repositories `developmental` branch.
- [x] I [signed off](https://docs.pi-hole.net/guides/github/how-to-signoff/) all commits. Pi-hole enforces the [DCO](https://docs.pi-hole.net/guides/github/dco/) for all contributions
- [x] I [signed](https://docs.github.com/en/authentication/managing-commit-signature-verification/signing-commits) all my commits. Pi-hole requires signatures to verify authorship
- [x] I have read the above and my PR is ready for review.

### CODE REVIEW

**Review by yubiuser — COMMENTED** [review]
I'm a bit confused about `TOML_UTF8` and the comments in the text. When this flag is set (new default), then the `toml_utf8_to_ucs()` is used converting the stings to ASCII + UCS. But this contradicts the changes in `toml_writer` and the inital comment in `config.h`.

File: src/config/toml_writer.c [hunk 1]
```diff
@@ -41,15 +41,17 @@ bool writeFTLtoml(const bool verbose)
 		log_info("Writing config file");
 
 	// Write header
-	fputs("# This file is managed by pihole-FTL\n#\n", fp);
-	fputs("# Do not edit the file while FTL is\n", fp);
```
Comment (yubiuser) [inline src/config/toml_writer.c:45]: Why did you remove this and the following line? 
I'm not sure if users understand that this is meant with the new "This file is managed by pihole-FTL"

Comment (DL6ER) [reply src/config/toml_writer.c:45]: Because it is explicitly supported by now. FTL actively monitors the file and reloads it on any changes. This is also how `pihole-FTL --config ...` works behind the scenes: It modifies the file on disk and the currently running FTL instance sees that and picks up the changes.

**Review by yubiuser — APPROVED** [review]
